vendor/engines/saml_authentication/README.md

```diff
@@ -0,0 +1,80 @@
+# SamlAuthentication Engine
+
+NUcore uses [Devise](https://github.com/plataformatec/devise) for authentication,
+and uses the [`saml_authenticatable`](https://github.com/apokalipto/devise_saml_authenticatable)
+module for SAML integration.
+
+## Enabling SAML Authentication
+
+Add the following line to your `Gemfile` if it is not already present:
+
+```ruby
+gem "saml_authentication", path: "vendor/engines/saml_authentication"`
+```
+
+In `config/settings.yml`, uncomment the `saml` section and adjust the settings
+as necessary to connect with your Identity Provider (IdP).
+
+```yaml
+saml:
+  idp_metadata: "https://websso.example.com/idp/metadata"
+  certificate_file: path/to/file.p12
+  attribute_map:
+    "PersonImmutableID": "username"
+    "User.email": "email"
+    "User.FirstName": "first_name"
+    "User.LastName": "last_name"
+```
+
+* `idp_metadata`: The URL for your IdP's metadata. This should be provided to you by the IdP.
+* `certificate_file` (Optional): A `.p12` certificate file for signing your requests.
+  _Do not check this in to version control_
+* `attribute_map`: A mapping from the IdP's attributes to the `User`'s attributes.
+  `username` and `email` are absolutely required while entries for `first_name`
+  and `last_name` are recommended.
+
+## Handling Users
+
+When a user logs in one of three things will happen (in this order):
+
+1. If the username field (as specified in `attribute_map`) matches an existing
+   NUcore `User`, they will be logged in. Other attributes coming from the SAML
+   exchange (`email`, `first_name`, `last_name`) will be updated.
+2. If the SAML user's email address matches an existing `User`'s email address,
```

Comment:
If the username matches but the email doesn't, is the password not cleared?

Reply:
If the username matches, the password should not be there.

Reply:
Okay. Might be nice to clarify this in the readme a bit--it's pretty clear in the test example, just a little confusing here.